### PR TITLE
Adding a bunch of new examples

### PR DESCRIPTION
**Pull request by chaithyagr**
I have added following new examples:
1) Multiresolution learning with finufft backend without density compensation. This should be runnable on binder!
2) Learning for multicoil data.
3) Learning of straight line readouts (this is a learning example where the trajectory is parametrized.

### CODE REVIEW

File: examples/GPU/example_learn_samples_multicoil.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+for f in image_files:
+    try:
+        os.remove(f)
+    except OSError:
+        continue
+# don't raise errors from pytest. This will only be executed for the sphinx gallery stuff
+try:
+    final_dir = (
+        Path(os.getcwd()).parent.parent
```
Comment (paquiteau) [inline examples/GPU/example_learn_samples_multicoil.py:191]: you probably want to use `__file__` instead. You don't have the guarantee that you are running from the example directory.

**Review by paquiteau — CHANGES_REQUESTED** [review]
Nice work! I recommend adding more comments everywhere to help (rule of thumb: as many comments/explanations as lines of code). I know this is tedious, but I guess an LLM can help with that ;) 


Maybe we should think about putting these examples in a sub-gallery?

**Review by paquiteau — COMMENTED** [review]
Awesome example ! 

I think we should add the equations explaining the autodiff aspects in the docs (`Explanation` rubric is perfect for that), and refer to it in the example.

File: examples/example_learn_samples_multires.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+.. math::
+    \mathbf{\hat{K}} =  arg \min_{\mathbf{K}} ||  \mathcal{F}_\mathbf{K}^* D_\mathbf{K} \mathcal{F}_\mathbf{K} \mathbf{x} - \mathbf{x} ||_2^2
+    
+where :math:`\mathcal{F}_\mathbf{K}` is the forward NUFFT operator and :math:`D_\mathbf{K}` is the density compensators for trajectory :math:`\mathbf{K}`,  :math:`\mathbf{x}` is the MR image which is also the target image to be reconstructed.
+
+Additionally, in-order to converge faster, we also learn the trajectory in a multi-resolution fashion. This is done by upscaling the trajectory by a factor of 2 and interpolating the control points. This is done for 5 iterations in this example.
+
+.. note::
+    This example can run on a binder instance as it is purely CPU based backend (finufft).
```
Comment (paquiteau) [inline examples/example_learn_samples_multires.py:21]:
```suggestion
    This example can run on a binder instance as it is purely CPU based backend (finufft), and is restricted to a 2D single coil toy case.
```

**Review by paquiteau — CHANGES_REQUESTED** [review]

File: examples/GPU/example_learn_straight_line_readouts.py
[the remark this answers is not in the document]
Comment (chaithyagr) [reply examples/GPU/example_learn_straight_line_readouts.py:1]: I removed density compensation which is causing the issue. I will add it back...

File: examples/example_learn_samples_multires.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+def plot_state(axs, mri_2D, traj, control_points, recon, loss=None, save_name=None):
+    axs = axs.flatten()
+    axs[0].imshow(np.abs(mri_2D[0]), cmap="gray")
+    axs[0].axis("off")
+    axs[0].set_title("MR Image")
+    axs[1].scatter(*traj.T, s=0.5)
+    axs[1].scatter(*control_points.T, s=1, color="r")
+    axs[1].legend(["Trajectory", "Control Points"])
+    axs[1].set_title("Trajectory")
```
[the remark this answers is not in the document]
Comment (chaithyagr) [reply examples/example_learn_samples_multires.py:122]: Ya possibly

File: examples/GPU/example_learn_samples_multicoil.py [hunk 4]
```diff
@@ -67,8 +67,9 @@ def __init__(self, inital_trajectory, n_coils, img_size=(256, 256)):
         )
         sample_points = inital_trajectory.reshape(-1, inital_trajectory.shape[-1])
         # A simple acquisition model simulated with a forward NUFFT operator. We dont need density compensation here.
+        # The trajectory is scaled by 2*pi for cufinufft backend.
         self.operator = get_operator("cufinufft", wrt_data=True, wrt_traj=True)(
-            sample_points,
+            sample_points*2*np.pi,
```
Comment (paquiteau) [inline examples/GPU/example_learn_samples_multicoil.py:72]: This is a no-op, the trajectory is going to go through `proper_trajectory` for doing the scaling.

File: examples/GPU/example_learn_samples_multicoil.py [hunk 5]
```diff
@@ -88,9 +89,10 @@ def __init__(self, inital_trajectory, n_coils, img_size=(256, 256)):
 
     def forward(self, x):
         """Forward pass of the model."""
-        # Update the trajectory in the NUFFT operator.
+        # Update the trajectory in the NUFFT operator. 
+        # The trajectory is scaled by 2*pi for cufinufft backend.
         # Note that the re-computation of density compensation happens internally.
-        self.operator.samples = self.trajectory.clone()
+        self.operator.samples = self.trajectory.clone()*2*np.pi
         self.sense_op.samples = self.trajectory.clone()
```
Comment (paquiteau) [inline examples/GPU/example_learn_samples_multicoil.py:96]: Same here

**Review by paquiteau — CHANGES_REQUESTED** [review]

**Comment by paquiteau** [comment]
LGTM !

**Pull request merged**